Cabal/Distribution/Simple/Utils.hs

```diff
@@ -810,6 +817,63 @@ rawSystemStdout verbosity path args = withFrozenCallStack $ do
     die errors
   return output
 
+-- | Represents either textual or binary data passed via I/O functions
+-- which support binary/text mode
+--
+-- @since 2.2.0
+data IOData = IODataText    String
+              -- ^ How Text gets encoded is usually locale-dependent.
+            | IODataBinary  BS.ByteString
+              -- ^ Raw binary which gets read/written in binary mode.
+
+-- | Test whether 'IOData' is empty
+--
+-- @since 2.2.0
+ioDataNull :: IOData -> Bool
+ioDataNull (IODataText s) = null s
+ioDataNull (IODataBinary b) = BS.null b
+
+instance NFData IOData where
+    rnf (IODataText s) = rnf s
+#if MIN_VERSION_bytestring(0,10,0)
+    rnf (IODataBinary bs) = rnf bs
+#else
+    rnf (IODataBinary bs) = rnf (BS.length bs)
+#endif
+
+data IODataMode = IODataModeText | IODataModeBinary
+
+-- | 'IOData' Wrapper for 'hGetContents'
+--
+-- __Note__: This operation uses lazy I/O. Use 'NFData' to force all
+-- data to be read and consequently the internal file handle to be
+-- closed.
+--
+-- @since 2.2.0
+ioDataHGetContents :: Handle -> IODataMode -> IO IOData
```

Comment:
Maybe move `IOData` & friends to a separate module and import that qualified instead of using a prefix? Like the existing `BS.`/`LBS.` precedent.